File: abc398a/abc398a.hs
{-# LANGUAGE OverloadedStrings #-}

main :: IO ()
main = solve <$> readLn >>= putStrLn

solve :: Int -> String
solve n = mn ++ eq ++ mn
  where
    mn = replicate ((n - 1) `div` 2) '-'
    eq = replicate (2 - n `mod` 2) '='

File: abc398a/abc398a.ml
open Printf
open Scanf

let solve n =
  if n mod 2 = 1 then String.init n (fun i -> if i = n / 2 then '=' else '-')
  else String.init n (fun i -> if i = n / 2 || i = n / 2 - 1 then '=' else '-')

let () =
  scanf "%d " solve |> printf "%s\n"
